harvester_e2e_tests/integration/test_rancher_integration.py

```diff
@@ -0,0 +1,365 @@
+                f"Got error: {code}, {data}"
+            )
+
+        # teardown
+        rancher_api_client.mgmt_clusters.delete(rke2_cluster_name)
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = rancher_api_client.mgmt_clusters.get(rke2_cluster_name)
+            if code == 404:
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"RKE2 MgmtCluster {rke2_cluster_name} can't be deleted \
+                    with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+    @pytest.mark.rke1
+    def test_create_rke1(self, rancher_api_client, unique_name, harvester_cloud_credential,
+                         rke1_cluster_name, focal_image, vlan_network, k8s_version, wait_timeout,
+                         rancher_wait_timeout):
+        code, data = rancher_api_client.node_templates.create(
+            name=unique_name,
+            cpus=2,
+            mems=4,
+            disks=40,
+            image_id=focal_image['id'],
+            network_id=vlan_network['id'],
+            ssh_user=focal_image['ssh_user'],
+            cloud_credential_id=harvester_cloud_credential['id'],
+            user_data=(
+                "#cloud-config\n"
+                "password: test\n"
+                "chpasswd:\n"
+                "    expire: false\n"
+                "ssh_pwauth: true\n"
+            ),
+        )
+        assert 201 == code, (
+            f"Failed to create NodeTemplate {unique_name} with error: {code}, {data}"
+        )
+
+        node_template_id = data['id']
+
+        code, data = rancher_api_client.clusters.create(rke1_cluster_name, k8s_version)
+        assert 201 == code, (
+            f"Failed to create cluster {rke1_cluster_name} with error: {code}, {data}"
+        )
+        cluster_id = data['id']
+
+        code, data = rancher_api_client.node_pools.create(
+            cluster_id=cluster_id,
+            node_template_id=node_template_id,
+            hostname_prefix=unique_name
+        )
+        assert 201 == code, (
+            f"Failed to create NodePools for cluster {cluster_id} with error: {code}, {data}"
+        )
+
+        endtime = datetime.now() + timedelta(seconds=rancher_wait_timeout)
+        while endtime > datetime.now():
+            code, data = rancher_api_client.mgmt_clusters.get(cluster_id)
+            if code == 200 and data.get('status', {}).get('ready', False):
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"RKE1 MgmtCluster {cluster_id} can't be ready \
+                    with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+        # teardown
+        rancher_api_client.mgmt_clusters.delete(cluster_id)
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
```

Comment:
I'm thinking maybe we can just use `rancher_wait_timeout` for all waiting cases :thinking: